src/bp/nlu-core/entities/microsoft-extractor/microsoft-extractor.test.ts

```diff
@@ -0,0 +1,263 @@
+            unit: 'phonenumber'
+          }
+        }
+      ],
+      [
+        {
+          confidence: 1,
+          type: 'ip',
+          value: '135.19.84.102',
+          start: 19,
+          end: 32,
+          metadata: {
+            source: '135.19.84.102',
+            entityId: 'system.ip',
+            extractor: 'system',
+            unit: 'ip'
+          }
+        }
+      ],
+      [
+        {
+          confidence: 1,
+          type: 'mention',
+          value: '@pedro',
+          start: 8,
+          end: 14,
+          metadata: {
+            source: '@pedro',
+            entityId: 'system.mention',
+            extractor: 'system',
+            unit: 'mention'
+          }
+        }
+      ],
+      [
+        {
+          confidence: 1,
+          type: 'hashtag',
+          value: '#sport',
+          start: 24,
+          end: 30,
+          metadata: {
+            source: '#sport',
+            entityId: 'system.hashtag',
+            extractor: 'system',
+            unit: 'hashtag'
+          }
+        }
+      ],
+      [
+        {
+          confidence: 1,
+          type: 'email',
+          value: 'hello@helloworld.com',
+          start: 25,
+          end: 45,
+          metadata: {
+            source: 'hello@helloworld.com',
+            entityId: 'system.email',
+            extractor: 'system',
+            unit: 'email'
+          }
+        }
+      ],
+      [
+        {
+          confidence: 1,
+          type: 'url',
+          value: 'www.thecuteboys.com',
+          start: 9,
+          end: 28,
+          metadata: {
+            source: 'www.thecuteboys.com',
+            entityId: 'system.url',
+            extractor: 'system',
+            unit: 'url'
+          }
+        }
+      ]
+    ]
+
+    const examples = [
+      'мой номер телефона +33 6 66 66 66 66',
+      'Можете попробовать 135.19.84.102, пожалуйста?',
+      'Привет, @pedro, можешь помочь?',
+      'Вы смотрели новый канал #sport? Это потрясающе',
+      'пн электронная почта est hello@helloworld.com',
+      'Мой сайт www.thecuteboys.com, пожалуйста, напишите отзыв'
+    ]
+    const res = await microsoft.extractMultiple(examples, 'ru')
+    res
+      .map((val, idx) => [val, results[idx]])
+      .forEach(([prem, hyp]) => {
+        expect(prem).toEqual(hyp)
+      })
+  })
+
+  test('returns as many results as n examples with single batch', async () => {
+    const examples = ['this is one', 'this is two', 'this is three']
+    const res = await microsoft.extractMultiple(examples, 'en')
+    expect(res.length).toEqual(examples.length)
+  })
+
+  test('returns as many results as n examples with multiple batches', async () => {
+    const examples = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '0', '11', '12', '13', '14']
+    const res = await microsoft.extractMultiple(examples, 'en')
+    expect(res.length).toEqual(examples.length)
+  })
+
+  test('good results for each examples', async () => {
```

Comment:
One test like this is fine and enough. If we want to do more testing like this, we'll do it with a benchmark and a dataset.